lib/streamlit/delta_generator.py

```diff
@@ -903,3 +923,9 @@ def _enqueue_message(msg: ForwardMsg_pb2.ForwardMsg) -> None:
         msg.delta.fragment_id = ctx.current_fragment_id
 
     ctx.enqueue(msg)
+
+
+def _writes_directly_to_sidebar(dg: DG) -> bool:
+    in_sidebar = sidebar_dg in set(dg._anscestors)
```

Comment:
This could also be `dg._root_container = RootContainer.SIDEBAR`, right?

Reply:
Ah, yeah. I think the code to check for that is a bit clunkier, but it probably works slightly better with tests that don't necessarily use the same `sidebar_dg` we define in this file. Changed